src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -3704,10 +3704,18 @@ def get_vmss(cmd, resource_group_name, name, instance_id=None, include_user_data
 def _check_vmss_hyper_v_generation(cli_ctx, vmss):
     hyper_v_generation = get_hyper_v_generation_from_vmss(
         cli_ctx, vmss.virtual_machine_profile.storage_profile.image_reference, vmss.location)
-    if hyper_v_generation == "V1":
+    security_profile = vmss.virtual_machine_profile.security_profile
+    security_type = security_profile.security_type if security_profile else None
+
+    if hyper_v_generation == "V1" or (hyper_v_generation == "V2" and security_type is None):
         logger.warning("Trusted Launch security type is supported on Hyper-V Generation 2 OS Images. "
                        "To know more please visit "
                        "https://learn.microsoft.com/en-us/azure/virtual-machines/trusted-launch")
+    elif hyper_v_generation == "V2" and security_type == "ConfidentialVM":
+        from azure.cli.core.azclierror import InvalidArgumentValueError
+        raise InvalidArgumentValueError("{} is already configured with {}. "
+                                        "Security Configuration cannot be updated from ConfidentialVM to "
+                                        "TrustedLaunch.".format(vmss.name, security_type))
```

Comment:
I would like to confirm if this additional verification will cause a breaking change for customers?

Reply:
> I would like to confirm if this additional verification will cause a breaking change for customers?

I don't think it would be a breaking change. The backend blocked the request and returned a `azure.core.exceptions.ResourceExistsError` in my test. 
![image](https://github.com/Azure/azure-cli/assets/32201005/d9c70a02-a2d9-47be-93a8-465817fde3ce)
I will verify this question with Ajay.